### PR TITLE
fix(25350): fix flakey token importing e2e test

### LINKED ISSUE #26350: Fix "Token List shows percentage increase for an ERC20 to..." flaky tests
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/95033/workflows/faea9f93-9600-4207-bb3c-2dc17eec663b/jobs/3536505/tests

### PR DESCRIPTION
**Pull request by DDDDDanica**
## **Description**
Captured in https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/95033/workflows/faea9f93-9600-4207-bb3c-2dc17eec663b/jobs/3536505/tests.

This test is flaky from not waiting for token price to be fetched and rendered properly. The simple fix would be 
- wait until token is imported after modal is closed 
- use `findVisibleElement` and the built in `timeout` to wait for element to be rendered properly 

Attached to this PR is a report [here](https://github.com/MetaMask/metamask-extension/pull/26351#issuecomment-2271840917) of running it 20 times after the fix and results in no flakiness.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26351?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/26350

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/tokens/token-list.spec.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ describe('Token List', function () {
     await driver.clickElement(
       '[data-testid="import-tokens-modal-import-button"]',
     );
+    await driver.findElement({ text: 'Token imported', tag: 'h6' });
```
Comment (DDDDDanica) [inline test/e2e/tests/tokens/token-list.spec.ts:45]: wait until token modal is closed and token is properly imported before asserting

File: test/e2e/tests/tokens/token-list.spec.ts [hunk 2]
```diff
@@ -155,40 +156,30 @@ describe('Token List', function () {
       async ({ driver }: { driver: Driver }) => {
         await unlockWallet(driver);
         await importToken(driver);
-        await driver.delay(500);
```
Comment (DDDDDanica) [inline test/e2e/tests/tokens/token-list.spec.ts:158]: removed with better `await` [here](https://github.com/MetaMask/metamask-extension/pull/26351/files#r1705837182)

File: test/e2e/tests/tokens/token-list.spec.ts [hunk 3]
```diff
@@ -155,40 +156,30 @@ describe('Token List', function () {
...
-          )
-        ).getText();
-        assert.equal(increaseValue, '+$50.00 ');
-
-        // check percentage increase balance for native token
-        const percentageIncreaseDecrease = await (
-          await driver.findElement(
-            `[data-testid="token-increase-decrease-percentage"]`,
-          )
-        ).getText();
```
Comment (DDDDDanica) [inline test/e2e/tests/tokens/token-list.spec.ts:189]: There is no timeout for this assertion, we need to give the element sometime to render and fetch.

**Comment by DDDDDanica** [comment]
[command_output.txt](https://github.com/user-attachments/files/16514061/command_output.txt)

**Review by desi — DISMISSED** [review]
LGTM

File: test/e2e/tests/tokens/token-list.spec.ts [hunk 4]
```diff
@@ -155,40 +156,30 @@ describe('Token List', function () {
       async ({ driver }: { driver: Driver }) => {
         await unlockWallet(driver);
         await importToken(driver);
-        await driver.delay(500);
 
         // Verify native token increase
         const testIdNative = `token-increase-decrease-percentage-${zeroAddress()}`;
 
         // Verify native token increase
         const testId = `token-increase-decrease-percentage-${tokenAddress}`;
```
Comment (danjm) [inline test/e2e/tests/tokens/token-list.spec.ts:165]: We have a recommended pattern for when we need to assert data that might be updated asynchronously. It is best if we have explicit assertions (separate from the `findElement` call) and it is necessary to also include expected text in the find element call. See the docs for the recommended pattern: https://github.com/MetaMask/contributor-docs/blob/main/docs/e2e/extension-e2e-guidelines.md#guidelines-3

Could you update it to follow that sort of pattern and maintain the use of explicit assert calls?

**Review by danjm — CHANGES_REQUESTED** [review]

File: test/e2e/tests/tokens/token-list.spec.ts (hunk 4 above)
Comment (DDDDDanica) [reply test/e2e/tests/tokens/token-list.spec.ts:184]: Thanks for linking the pattern, it is indeed needed for text assertion, changed here [8d5843a](https://github.com/MetaMask/metamask-extension/pull/26351/commits/8d5843a5d07997aedcd0a8d4c82bcaa79b533d78)

**Comment by DDDDDanica** [comment]
Updated after [8d5843a](https://github.com/MetaMask/metamask-extension/pull/26351/commits/8d5843a5d07997aedcd0a8d4c82bcaa79b533d78)
[command_output.txt](https://github.com/user-attachments/files/16550858/command_output.txt)

**Review by danjm — APPROVED** [review]
LGTM!

**Review by benjisclowder — APPROVED** [review]
LGTM!

**Pull request merged**